mgmtsystem/hawkular.py

```diff
@@ -766,6 +788,372 @@ def delete_resource(self, feed_id, resource_id):
+        HawkularService.__init__(self, hostname=hostname, port=port, protocol=protocol,
+                                 auth=auth, tenant_id=tenant_id, entry="hawkular/metrics")
+
+    _stats_available = {
+        'num_server': lambda self: len(self.list_server()),
+        'num_domain': lambda self: len(self.list_domain()),
+        'num_deployment': lambda self: len(self.list_server_deployment()),
+        'num_datasource': lambda self: len(self.list_server_datasource()),
+        'num_messaging': lambda self: len(self.list_messaging()),
+    }
+
+    def list_feed(self):
+        """Returns list of feeds"""
+        entities = []
+        entities_j = self._get('strings/tags/module:inventory,feed:*')
+        if entities_j and entities_j['feed']:
+            for entity_j in entities_j['feed']:
+                entities.append(Feed(entity_j, CanonicalPath('/f;{}'.format(entity_j))))
+        return entities
+
+    def list_server(self, feed_id=None):
+        """Returns list of middleware servers.
+
+          Args:
+            feed_id: Feed id of the resource (optional)
+        """
+        resources = self.list_resource(feed_id=feed_id,
+                                       resource_type_id='WildFly Server',
+                                       include_data=True)
+        resources.extend(self.list_resource(
+            feed_id=feed_id,
+            resource_type_id='Domain WildFly Server',
+            list_children=True,
+            include_data=True))
+        servers = []
+        for resource in resources:
+            servers.append(Server(resource.id, resource.name, resource.path, resource.data))
+        return servers
+
+    def list_domain(self, feed_id=None):
+        """Returns list of middleware domains.
+
+          Args:
+            feed_id: Feed id of the resource (optional)
+        """
+        resources = self.list_resource(feed_id=feed_id,
+                                       resource_type_id='Domain Host',
+                                       list_children=True,
+                                       include_data=True)
+        domains = []
+        if resources:
+            for resource in resources:
+                domains.append(Domain(resource.id, resource.name, resource.path, resource.data))
+        return domains
+
+    def list_server_group(self, feed_id):
+        """Returns list of middleware domain's server groups.
+
+          Args:
+            feed_id: Feed id of the resource (optional)
+        """
+        resources = self.list_resource(feed_id=feed_id,
+                                       resource_type_id='Domain Server Group',
+                                       list_children=True,
+                                       include_data=True)
+        server_groups = []
+        if resources:
+            for resource in resources:
+                server_groups.append(ServerGroup(resource.id, resource.name,
```

Comment:
@RonnyPfannschmidt can we leave this maintenance for later? it requires changes on the old API as well, in this case